### DISCUSSION TITLE
Help! Cannot push to Git because of .pack files being too large in React app

### DISCUSSION THREAD

**Discussion by rodricksnyc** [community/community#33067] [Actions]
Hi! Can anyone help me? I cannot push to Git because of these types of errors:

File node_modules/.cache/default-development/1.pack is 117.65 MB; this exceeds GitHub's file size limit of 100.00 MB

I tried using this documentation, which is what was recommended:
https://git-lfs.github.com/

I ran
 git lfs install
git lfs track "*.pack" 
git add .gitattributes

But I still could not push because of the file size. Does anyone know what I should do? Thanks!

**Comment by airtower-luna** [comment]
Moving the file to LFS doesn't remove it from the repository history, so it still gets pushed. You need to also [remove the file from repository history](https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-large-files-on-github#removing-files-from-a-repositorys-history).

However, the path smells wrong. Pushing a `.cache` directory to Git looks questionable at best, LFS or not. Most likely instead of using LFS you should remove the whole cache from your repository index and add it to `.gitignore`, so you don't waste repository space and worse (because expensive) LFS space on a cache.

**Reply by rodricksnyc** [reply]
I have tried using that link and even thought it shows that I deleted the directory form my project, it still shows these error messages when I try to push:
remote: error: File node_modules/.cache/default-development/0.pack is 138.60 MB; this exceeds GitHub's file size limit of 100.00 MB
remote: error: File node_modules/.cache/default-development/1.pack is 117.65 MB; this exceeds GitHub's file size limit of 100.00 MB
remote: error: GH001: Large files detected. You may want to try Git Large File Storage - https://git-lfs.github.com.

I am really not sure what to do. I added node_modules to .gitignore and that does not worl. I've never seen this before with a repo....

**Reply by airtower-luna** [reply]
Which part of the description have you used? That your repository still contains the large files sounds like you might've looked only at the first part, about changing the *latest* commit. You'll probably need to go further back (the section below). It's hard to say more without knowing the exact steps you took.
